plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -27,65 +27,140 @@
     private final static Logger LOGGER = Logger.getLogger(VaultSecretSource.class.getName());
     private Map<String, String> secrets = new HashMap<>();
 
+    private static final String CASC_VAULT_FILE = "CASC_VAULT_FILE";
+    private static final String CASC_VAULT_PW = "CASC_VAULT_PW";
+    private static final String CASC_VAULT_USER = "CASC_VAULT_USER";
+    private static final String CASC_VAULT_URL = "CASC_VAULT_URL";
+    private static final String CASC_VAULT_MOUNT = "CASC_VAULT_MOUNT";
+    private static final String CASC_VAULT_TOKEN = "CASC_VAULT_TOKEN";
+    private static final String CASC_VAULT_APPROLE = "CASC_VAULT_APPROLE";
+    private static final String CASC_VAULT_APPROLE_SECRET = "CASC_VAULT_APPROLE_SECRET";
+    private static final String CASC_VAULT_NAMESPACE = "CASC_VAULT_NAMESPACE";
+    private static final String CASC_VAULT_ENGINE_VERSION = "CASC_VAULT_ENGINE_VERSION";
+    private static final String CASC_VAULT_PATHS = "CASC_VAULT_PATHS";
+    private static final String CASC_VAULT_PATH = "CASC_VAULT_PATH"; // TODO: deprecate!
+
+
     public VaultSecretSource() {
-        String vaultFile = System.getenv("CASC_VAULT_FILE");
+        Optional<String> vaultFile = Optional.ofNullable(System.getenv(CASC_VAULT_FILE));
         Properties prop = new Properties();
-        if (vaultFile != null) {
-            try (FileInputStream input = new FileInputStream(vaultFile)) {
-                prop.load(input);
-                if (prop.isEmpty()) {
-                    LOGGER.log(Level.WARNING, "Vault secret file is empty");
-                }
-            } catch (IOException ex) {
-                LOGGER.log(Level.WARNING, "Failed to load Vault secrets from file", ex);
+        vaultFile.ifPresent(file -> readPropertiesFromVaultFile(file, prop));
+
+        // Parse variables
+        Optional<String> vaultPw = getVariable(CASC_VAULT_PW, prop);
+        Optional<String> vaultUser = getVariable(CASC_VAULT_USER, prop);
+        Optional<String> vaultUrl = getVariable(CASC_VAULT_URL, prop);
+        Optional<String> vaultMount = getVariable(CASC_VAULT_MOUNT, prop);
+        Optional<String> vaultToken = getVariable(CASC_VAULT_TOKEN, prop);
+        Optional<String> vaultAppRole = getVariable(CASC_VAULT_APPROLE, prop);
+        Optional<String> vaultAppRoleSecret = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
+        Optional<String> vaultNamespace = getVariable(CASC_VAULT_NAMESPACE, prop);
+        Optional<String> vaultEngineVersion = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
+        Optional<String[]> vaultPaths = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
+                .map(Optional::of)
+                .orElse(getCommaSeparatedVariables(CASC_VAULT_PATH, prop)); // TODO: deprecate!
+
+        // Check mandatory variables are set
+        if(!vaultUrl.isPresent()) {
+            LOGGER.log(Level.WARNING, "Mandatory variable {0} not set. Cannot fetch from vault.", CASC_VAULT_URL);
```

Comment:
Uff, this class gets called on startup/re-configure. I think it should just return silently like it did before :smile:
Some users might not use vault

Reply:
Removed it :+1: